internal/agent-api/types.go

```diff
@@ -71,32 +75,37 @@ type DeployRequest struct {
 	Errors []error `json:"errors,omitempty"`
 }
 
-func (w *DeployRequest) Validate() bool {
-	w.Errors = make([]error, 0)
+// Returns true if the run request supports trigger subjects
+func (request *DeployRequest) SupportsTriggerSubjects() bool {
+	return (strings.EqualFold(*request.WorkloadType, "v8") ||
+		strings.EqualFold(*request.WorkloadType, "wasm")) &&
+		len(request.TriggerSubjects) > 0
+}
+
+func (r *DeployRequest) Validate() bool {
+	r.Errors = make([]error, 0)
+
+	if r.WorkloadName == nil {
+		r.Errors = append(r.Errors, errors.New("workload name is required"))
+	}
+
+	if r.Hash == "" { // FIXME--- this should probably be checked against *string
+		r.Errors = append(r.Errors, errors.New("hash is required"))
+	}
 
-	if w.WorkloadName == nil {
-		w.Errors = append(w.Errors, errors.New("workload name is required"))
+	if r.TotalBytes == 0 { // FIXME--- this should probably be checked against *string
+		r.Errors = append(r.Errors, errors.New("total bytes is required"))
```

Comment:
errors has errors.Join since 1.20 and is probably the more idomatic way of combining errors now

Reply:
Much better, good call.